src/ol/control/control.js

```diff
@@ -121,3 +124,18 @@ ol.control.Control.prototype.setMap = function(map) {
     map.render();
   }
 };
+
+
+/**
+ * This function is used to set a target element for the control. This
+ * function will have no effect if it is called after the control has
```

Comment:
What about
```js
ol.control.Control.prototype.setTarget = function(target) {
  var map = this.map_;
  this.setMap(null);
  this.target_ = target;
  this.setMap(map);
};
```
If that does not work, you could at least add an assertion to ensure that the map is not set yet.